src/main/java/setup/BaseTest.java

```diff
@@ -1,67 +1,166 @@
+        capabilities.setCapability("udid", udid);
 
-        if(app.endsWith(".apk")) capabilities.setCapability("app", (new File(app)).getAbsolutePath());
+        if (app.endsWith(".apk")) {
+            capabilities.setCapability("app", (new File(app)).getAbsolutePath());
+        }
 
         capabilities.setCapability("browserName", browserName);
-        capabilities.setCapability("chromedriverDisableBuildCheck","true");
+        capabilities.setCapability("chromedriverDisableBuildCheck", "true");
+
+        // Capabilities for test of Android native app on EPAM Mobile Cloud
+        capabilities.setCapability("appPackage", appPackage);
+        capabilities.setCapability("appActivity", appActivity);
+
+        // Capabilities for test of iOS native app on EPAM Mobile Cloud
+        capabilities.setCapability("bundleId", bundleId);
+        //        if(platformName.equals("iOS")) capabilities.setCapability("automationName","XCUITest");
 
         try {
-            appiumDriver = new AppiumDriver(new URL(System.getProperty("ts.appium")), capabilities);
+
+            String url = "https://" + System.getenv("USERNAMEMOBIT") + ":" + URLEncoder.encode(System.getenv("TOKEN"))
+                + "@app.mobitru.com/wd/hub";
+            System.out.println("url:" + url + capabilities);
+            appiumDriver = new AppiumDriver(new URL(url), capabilities);
         } catch (MalformedURLException e) {
             e.printStackTrace();
         }
 
         // Timeouts tuning
-        appiumDriver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
-
+        appiumDriver.manage().timeouts().implicitlyWait(waitTime, TimeUnit.SECONDS);
     }
 
     private void setPageObject(String appType, AppiumDriver appiumDriver) throws Exception {
         po = new PageObject(appType, appiumDriver);
     }
 
+    protected void waitUntilPageLoad() {
+        new WebDriverWait(getDriver(), waitTime).until(
+            wd -> ((JavascriptExecutor) wd).executeScript(isPageReadyScript).equals(isPageReadyScriptResult)
+        );
+    }
+
+    //Getters for pages
+    public LogInPage getLogInPage() {
+        if (logInPage == null) {
+            return new LogInPage(appiumDriver);
+        } else {
+            return logInPage;
+        }
+    }
+
+    public RegisterPage getRegisterPage() {
+        if (registerPage == null) {
+            return new RegisterPage(appiumDriver);
+        } else {
+            return registerPage;
+        }
+    }
+
+    public BudgetActivityPage getBudgetActivityPage() {
+        if (webDriverWait == null) {
+            return new BudgetActivityPage(appiumDriver);
+        } else {
+            return budgetActivityPage;
+        }
+    }
+
+    public AppiumFluentWait getWebDriverWait() {
+        if (webDriverWait == null) {
+            return new AppiumFluentWait(appiumDriver);
+        } else {
+            return webDriverWait;
+        }
+    }
 
+    public void acceptHungaryCookie(GooglePage page) {
```

Comment:
It's better to keep this method on the page layer, as well as other methods that don't relate to tests logic, but to pages

Reply:
Fixed